src/peergos/shared/user/UserContext.java

```diff
@@ -518,30 +520,33 @@ public boolean isJavascript() {
                                                 newPublicSigningKey,
                                                 tid),
                                         network.dhtClient
-                                ).thenCompose(newSignerHash -> userData.runWithLock(wd -> wd.props.changeKeys(
-                                        signer,
-                                        new SigningPrivateKeyAndPublicHash(newSignerHash, updatedUser.getUser().secretSigningKey),
-                                        wd.hash,
-                                        updatedUser.getBoxingPair().publicBoxingKey,
-                                        existingUser.getRoot(),
-                                        updatedUser.getRoot(),
-                                        newAlgorithm,
-                                        network
-                                )).thenCompose(writerData -> {
-                                    SigningPrivateKeyAndPublicHash newUser =
-                                            new SigningPrivateKeyAndPublicHash(newSignerHash, updatedUser.getUser().secretSigningKey);
-                                    return network.coreNode.getChain(username).thenCompose(existing -> {
-                                        List<Multihash> storage = existing.get(existing.size() - 1).claim.storageProviders;
-                                        List<UserPublicKeyLink> claimChain = UserPublicKeyLink.createChain(signer, newUser, username, expiry, storage);
-                                        return network.coreNode.updateChain(username, claimChain)
-                                                .thenCompose(updatedChain -> {
-                                                    if (!updatedChain)
-                                                        throw new IllegalStateException("Couldn't register new public keys during password change!");
-
-                                                    return UserContext.ensureSignedUp(username, newPassword, network, crypto);
-                                                });
-                                    });
-                                }));
+                                ).thenCompose(newSignerHash ->
```

Comment:
if only indentation has changed here can we revert it?

Reply:
nope, the code structure has changed.